frappe/desk/page/modules/modules.js

```diff
@@ -134,8 +134,9 @@ frappe.pages['modules'].on_page_load = function(wrapper) {
 				}
 
 				if(item.type==="page" || item.type==="help" ||
-					(item.doctype && frappe.model.can_read(item.doctype))) {
-						item.shown = true;
+					(item.doctype && (frappe.model.can_read(item.doctype) ||
+						(item.type=="report" && frappe.model.can_view_report(item.doctype))) )) {
```

Comment:
Why report visibility is still based on doctype? Report has now it's own permission, should based on that.